File: 9.20/OC封装网络请求工具类/OC封装网络请求工具类/NetWorkTool.m
//
//  NetWorkTool.m
//  OC封装网络请求工具类
//
//  Created by codygao on 16/9/20.
//  Copyright © 2016年 HM. All rights reserved.
//


#import "NetWorkTool.h"


@implementation NetWorkTool
//创建全局单例接入点
+(instancetype)sharedtool{
    
    static NetWorkTool *tool;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        tool = [[self alloc] init];
        tool.responseSerializer.acceptableContentTypes = [tool.responseSerializer.acceptableContentTypes setByAddingObject:@"text/html"];
        
        
    });
    return  tool;
}


-(void)requesttype:(RequstType)type url:(NSString *)url parameters:(id) parameters callback:(void (^) (id response,NSError *errror))callback{
    
    if (type == GET){
    //封装GET类
        [self GET:url parameters:nil progress:nil success:^(NSURLSessionDataTask * _Nonnull task, id  _Nullable responseObject) {
            callback(responseObject, nil);
            
        } failure:^(NSURLSessionDataTask * _Nullable task, NSError * _Nonnull error) {
            callback(nil, error);
        }];
        
    }else{
    //封装Post类
    
        [self POST:url parameters:nil progress:nil success:^(NSURLSessionDataTask * _Nonnull task, id  _Nullable responseObject) {
            callback(responseObject,nil);
        } failure:^(NSURLSessionDataTask * _Nullable task, NSError * _Nonnull error) {
            callback(nil,error);
        }];
        
    }
    
    
}
@end

File: 9.20/swift封装网络请求工具类一/swift封装网络请求工具类一/NetworkTool.swift
//
//  NetworkTool.swift
//  swift封装网络请求工具类一
//
//  Created by codygao on 16/9/21.
//  Copyright © 2016年 HM. All rights reserved.
//

import UIKit
import AFNetworking
enum RequestType: Int {
    case GET
    case POST
}

class NetworkTool: AFHTTPSessionManager{
    
    //创建全局单例接入口
    static let sharedtool: NetworkTool = {
        
        let tool = NetworkTool()
        tool.responseSerializer.acceptableContentTypes?.insert("text/html")
          return tool
    }()
   
    
    func requestType(type: RequestType,url: String,parameter: Any?,callback: @escaping (Any?,Error?)->Void){
        
        //枚举要用.
        if type == .GET{
            get(url, parameters: nil, progress: nil, success: { (_, response) in
                callback(response, nil)
                }, failure: { (_, error) in
                    callback(nil,error)
            })
        }else{
            post(url, parameters: nil, progress: nil, success: { (_, response) in
                callback(response,nil)
                }, failure: { (_, error) in
                    callback(nil,error)
            })
        }
        
        
    }

}
